### PR TITLE
MM-33846 disallow cyclic shared channel invites

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary
This PR ensures that when a cluster has been invited to a shared channel, that cluster cannot then invite other organizations.  This also prevents any cyclic sync scenarios by ensuring only the home server for a channel can send invitations.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33846

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: app/shared_channel.go [hunk 1]
```diff
@@ -40,6 +40,23 @@ func (a *App) checkChannelIsShared(channelId string) error {
 	return nil
 }
 
+func (a *App) CheckCanInviteToSharedChannel(channelId string) error {
+	var err error
+	var sc *model.SharedChannel
+	if sc, err = a.GetSharedChannel(channelId); err != nil {
```
Comment (agnivade) [inline app/shared_channel.go:46]: I'd suggest breaking up the if-err clause, and remove the `var` declarations above.

**Review by agnivade — COMMENTED** [review]
Can we also add a test please? :)

File: app/shared_channel.go (hunk 1 above)
Comment (wiggin77) [reply app/shared_channel.go]: Fixed via https://github.com/mattermost/mattermost-server/pull/17155/commits/3d505410b1785518bda895015e632d5a873c14da  including unit tests.

**Review by agnivade — APPROVED** [review]
Hmm .. the new linter unfortunately caught some issues in this branch. I will send a PR to disable it. Looks like it will cause a lot of pain for others.

**Review by isacikgoz — APPROVED** [review]
LGTM 👍

**Pull request merged**